extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/language_server/BlocLanguageServer.kt

```diff
@@ -1,6 +1,97 @@
 package com.bloc.intellij_generator_plugin.language_server
 
+import com.bloc.intellij_generator_plugin.util.BlocPluginNotification
+import com.bloc.intellij_generator_plugin.util.CommandLineHelper
+import com.bloc.intellij_generator_plugin.util.VersionComparator
 import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
 import com.redhat.devtools.lsp4ij.server.OSProcessStreamConnectionProvider
+import java.util.concurrent.atomic.AtomicBoolean
 
-class BlocLanguageServer : OSProcessStreamConnectionProvider(GeneralCommandLine("bloc", "language-server"))
\ No newline at end of file
+class BlocLanguageServer(private val project: Project? = null) :
+    OSProcessStreamConnectionProvider(GeneralCommandLine("bloc", "language-server")) {
+
+    companion object {
+        private const val PROCESS_TIMEOUT_SECONDS: Long = 5
+        private const val TARGET_BLOC_TOOLS_VERSION = "0.1.0-dev.11"
+        private const val MIN_DART_VERSION = "3.7.0"
+    }
+
+    private val logger = Logger.getInstance(BlocLanguageServer::class.java)
+    private val minVersionAlertShowed = AtomicBoolean(false)
+    private val dartVersionRegex = Regex("Dart SDK version: ((?:[0-9]*\\.?)*)")
+
+    override fun start() {
+        if (project != null && checkRequirements(project)) {
```

Comment:
```suggestion
        if (project == null || !checkRequirements(project)) {
```

Isn't this logic slightly incorrect? I thought `checkRequirements` returns `true` if the requirements are met.

Reply:
Yeah it's an issue on my side